models/backup_capture.py

```
@@ -0,0 +1,389 @@
+                MIN(c.release_year)            AS release_year,
+                COUNT(DISTINCT c.capture_id)   AS version_count,
+                SUM(c.total_size)              AS total_size,
+                MAX(c.captured_at)             AS latest_captured_at,
+                MAX(c.pinned)                  AS has_pinned
+            FROM backup_capture c
+            JOIN backup_capture_key k ON k.capture_id = c.capture_id
+            WHERE {' AND '.join(where)}
+            GROUP BY k.slot_key
+            ORDER BY {order}
+            LIMIT ? OFFSET ?
+        '''
```

Ruff (0.16.0)

[error] Possible SQL injection vector through string-based query construction

(S608)